contracts/Distributor.sol

```diff
@@ -0,0 +1,250 @@
+    function registerPayeesForToken(address[] memory accounts_, IERC20 token_, uint256[] memory shares_) external onlyOwner {
+        require(accounts_.length == shares_.length, "Distributor: Mismatch of addresses and shares.");
+        for(uint256 i = 0; i < accounts_.length; i++ ){
+            _registerPayeesForTokens(accounts_[i], token_, shares_[i]);
+        }
+    }
+    
+    // @Desc: Payout/Claim eligible amount of ether from contract based on shares
+    function payoutETH(address payable account_) public {
+        address payable account = account_;
+        require(_sharesForETH[account_] > 0, "Distributor: Not registered to claim ETH.");
+
+        uint256 ethClaimable = claimableETH(account);
+
+        //Send ether to receipient via a low-level call function
+        require(address(this).balance >= ethClaimable, "Distributor: Insufficient ETH balance in contract.");
+        
+        _totalETHDistributed += ethClaimable;
+        _distributedETH[account] = ethClaimable;
+        (bool success,) = account.call{value: ethClaimable}("");
+        require(success, "Distributor: Unable to receive ETH from contract.");
+
+        emit ETHFundClaimed(account, ethClaimable);
+    }
+
+    function payoutToken(address account_, IERC20 token_) public {
+        address account = account_;
+        IERC20 token = token_;
+        require(_sharesForTokens[token][account] > 0, "Distributor: Not registered to claim ERC20 token.");
+
+        uint256 tokenAmountClaimable = claimableToken(account, token);
+
+        // Transfer ERC20 via transfer function from SafeERC20
+        require(token.balanceOf(address(this)) >= tokenAmountClaimable, "Distributor: Insufficient ERC20 token balance.");
+        
+        _totalTokenDistributed[token] += tokenAmountClaimable;
+        _distributedTokens[token][account] += tokenAmountClaimable;
+        SafeERC20.safeTransfer(token, account, tokenAmountClaimable);
+
+        emit TokenFundClaimed(account, token, tokenAmountClaimable);
+    }
+
+    // @Desc: Convenient One-Time Function call for owner to distribute ether from contract to all registered payees
+    function distributeDemAllETH() external onlyOwner {
+        uint totalPayees = numRegisteredForETH();
+        for(uint256 i = 0; i < totalPayees;){
+            if(_sharesForETH[_payeesForETH[i]] > 0) {
+                payoutETH(payable(_payeesForETH[i]));
+            }
+            unchecked {i++; }
+        }
+    }
+
+       // @Desc: Convenient One-Time Function call for owner to distribute a specified ERC20 Token from contract to all registered payees
+    function distributeDemAllToken(IERC20 token_) external onlyOwner {
+        IERC20 token = token_;
+        uint totalPayees = numRegisteredForToken(token);
+        for(uint256 i = 0; i < totalPayees;){
+            payoutToken(_payeesForToken[token][i], token);
+            unchecked {i++; }
+        }
+    }
```

Comment:
will revert if the share of one payee is 0

Reply:
It won't, because I am using a nested mapping to keep track of accounts registered under a particular token (means every account looped through will have shares > 0 for that token) itself